atlp-rwanda/e-comm-team-amigos-bn
# bg-change-user-role-structure

### What does this PR do?
This PR allow all endpoints using role to use **UserRole** and **Role** relations

### Description of Task to be completed?
Have the following middleware and endpoint , as well as tests use **UserRole** and **Role** relations. 

**Middleware**
- `authorize`
- `verifyToken`

**Endpoint**
- `GET /product/<product-id>`

- It also ensures that upon successfully signup, new user is given role of **Customer** by default using relations mentioned above.
- Lastly, it updates all the **tests** 

### How should this be manually tested?
In your e-comm-team-amigos-bn directory, open terminal and run `git fetch origin pull/48/head:bg-change-user-role-structure`

- Checkout to bg-change-user-role-structure
- Delete role in **User Schema**
- Run migrations (especialy a migration that removes role filed in user table);
- Run npm install
- Run npm run dev

In postman or swagger documentation do the following
- Make request on `POST /user/signup`
- Make request on `GET /user/profile`. You should see field **UserRoles** of an array of your roles. Which i by default **Customer**

### Any background context you want to provide?
- Have Node Js installed in your local machine
- Should have express server running
- Should have your .env file with neccessary variables as in .env.example

### Screenshoot

`Test Coverage`
![Screenshot (1501)](https://user-images.githubusercontent.com/55640083/229235201-66f45dcd-84ee-4906-a162-4c6939053621.png)
